src/intellisense/wrapper.rs

```diff
@@ -177,200 +175,169 @@ impl DxcCursor {
             let mut result: *mut *mut IDxcCursor = std::ptr::null_mut();
             let mut result_length: u32 = 0;
 
-            let mut children = vec![];
-            return_hr!(
+            check_hr!(
                 self.inner
                     .get_children(skip, max_count, &mut result_length, &mut result),
-                {
-                    for i in 0..result_length {
+                // get_children allocates a buffer to pass the result in.  Vec will
+                // free it on Drop.
+                Vec::from_raw_parts(result, result_length as usize, result_length as usize)
```

Comment:
I think this may be unsafe because `result` is from a different allocator then what `Vec::drop` will call.

Reply:
This is more inconvenient than thought because the tests use `CoTaskMemFree`, but the `GetChildren` function allocating these seem to override the thread allocator:

https://github.com/microsoft/DirectXShaderCompiler/blob/a8d9780046cb64a1cea842fa6fc28a250e3e2c09/tools/clang/tools/libclang/dxcisenseimpl.cpp#L836

I don't think this overrides what allocator is used by `CoTaskMemFree`, this is merely a [concept in DXC](https://github.com/microsoft/DirectXShaderCompiler/blob/a8d9780046cb64a1cea842fa6fc28a250e3e2c09/lib/DxcSupport/dxcmem.cpp)?

So, while it is possible to allocate things on a different `IMalloc` (we have a `SetMalloc` on `IDxcLibrary`) I don't think that's used for this particular list (`CoTaskMemAlloc`), and should hence be safely freed using `CoTaskMemFree` on Windows and `free` everywhere else?

Perhaps we should open up an issue in DXC to clarify this interface, and provide a convenient function to do the cleanup for us in their OS-dependent way, as we can't be guaranteed this never changes.

Reply:
The Windows CI didn't fail meaning I didn't misstype `pub use winapi::um::combaseapi::CoTaskMemFree;` nor forgot the feature flag, but I have not tested actually running this code on Windows yet.

Reply:
Yeah I think `CoTaskMemFree` here is fine because the DXC unit-tests also call `CoTaskMemFree` here.